### ISSUE TITLE
Open repo

### ISSUE THREAD

**Issue by roonyh** [roonyh/fossasia-gci-companion#2]
Allow users to open the repo of fossasia gci 15 [website](http://gci15.fossasia.org/) with the app. It should show if `origin` and `upstream` remotes are configured correctly.

**Comment by YagoGG** [comment]
Working on this issue...

**Comment by roonyh** [comment]
@YagoGG Great!

Here are some ideas: 
- Use [Dialog](http://electron.atom.io/docs/v0.36.0/api/dialog/) module. Its only available in the main process. (read about electron's renderer and main processes)
- To communicate between the main and renderer processes you can use the [ipcMain](http://electron.atom.io/docs/v0.36.0/api/ipc-main/) module.

Let me know if you have other ideas.

**Comment by YagoGG** [comment]
**Dialog** was exactly what I was looking for! Thanks!

**Comment by YagoGG** [comment]
I'm having some problems trying to run code after all the code [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) has finished running.
I need to use a variable in a `switch() case` structure, but that variable's content has been set inside the promise.
Chaining a `.then()` after the `Promise` won't work.

Here's a synthesis of what I'm trying to do:

``` javascript
Git.Repository.open(path)
        .then(function(repo) {
                // Generate myVar
        })

switch(myVar)
        case ...

```

If I attempt to run the code above, the `switch` won't work properly, because the part before is async, and `myVar` hasn't been set, yet.

Ideas?

**Comment by roonyh** [comment]
As you've said code inside the then block runs asynchronously. So there is no alternative. You have to use the switch case statement inside the then block. Why do you need to keep it outside?

You can make a PR with what you did so far and keep it open and work on it. Others can review and give you feedback that way.

**Comment by abdulhannanali** [comment]
@YagoGG If you are having a problem with understanding asynchronous logic, try out the guides on bluebirdjs.com , it's what nodegit uses internally.

**Comment by YagoGG** [comment]
Thanks you both! I finally sorted it out, I'll create my PR soon.

**Comment by roonyh** [comment]
@YagoGG good to hear. :)

**Comment by roonyh** [comment]
closed by #14
